### DISCUSSION TITLE
How to force TCP on a rtsp stream.

### DISCUSSION THREAD

**Discussion by BowenAV** [ant-media/Ant-Media-Server#6678] [Q&A]
I have a panoramic ONVIF compliant camera and want to use antmedia to provide a public 24/7 livestream. 

The camera works fine connected to my NVR but, when viewing through an RTSP URL or the IP camera option in AMS I get the video graying out. [An example from stackoverflow](https://stackoverflow.com/questions/23238295/force-ffmpeg-to-use-tcp-protocol-when-reading-a-rtsp-stream). 
I have narrowed it down to UDP being used instead of TCP (checked using VLC). the only problem is getting AMS to use TCP. If you have any suggestions please let me know, or shame me for not figuring it out.

Community Edition 2.11.3 20240906_1141 
ubuntu 24.04 (Desktop)

**Comment by lastpeony** [comment]
go to advanced app settings through web panel and set
 
`"rtspPullTransportType":"1"
`

The default value is 3, which is tcp(1) OR udp(2)
